src/mpid/common/thread/mpidu_thread_fallback.h

```diff
@@ -95,8 +95,62 @@ typedef MPL_thread_func_t MPIDU_Thread_func_t;
 - _context - A context (typically an object) of the critical section
 
 M*/
+
+/*M MPIDU_THREAD_CS_TRYENTER - Try enter a named critical section
+
+  Input Parameters:
++ _name - name of the critical section
+- _context - A context (typically an object) of the critical section
+- cs_acqt  - A flag that indicades whether the critical section was acquired
+
+M*/
+
+/*M MPIDU_THREAD_CS_EXIT - Exit a named critical section
+
+  Input Parameters:
++ _name - cname of the critical section
+- _context - A context (typically an object) of the critical section
+
+M*/
+
+/*M MPIDU_THREAD_CS_YIELD - Temporarily release a critical section and yield
+    to other threads
+
+  Input Parameters:
++ _name - cname of the critical section
+- _context - A context (typically an object) of the critical section
+
+  M*/
+
+/*M MPIDU_THREAD_CS_ENTER_ST - Stateful version of CS_ENTER
+ * It takes a MPIDU_thread_state_t variable to pass the state between ENTER and EXIT.
+  M*/
+
+/*M MPIDU_THREAD_CS_EXIT - Stateful version of CS_EXIT
+ * It takes a MPIDU_thread_state_t variable to pass the state between ENTER and EXIT.
+  M*/
+
+/* NOTE: this is currently only used by ch3:sock and only with GRANULARITY_GLOBAL */
```

Comment:
What is used only by `ch3:sock`?

Reply:
The stateful CS enter/exit. I see the comment is not as helpful I thought, deleted.